cms/src/builder/utils/kw.rs

```diff
@@ -90,210 +100,132 @@ impl KeyWrapAlgorithm {
-/// It currently implements:
-/// - try_new() - initialize a key wrapper with right sized depending on KeyWrapAlgorithm and key-to-wrap size
-/// - try_wrap() - wrap a key with the corresponding aes-key-wrap algorithms
-///
-/// # Note
-/// For convenience KeyWrapper can:
-/// - yield the inner wrapping-key as a mutable reference (e.g. to use with a KDF)
-/// - convert to Vec<u8> to obtain Owned data to the wrapped key
-#[derive(Debug, Clone, Copy)]
-pub(in crate::builder) struct KeyWrapper {
-    /// Wrapping key
-    wrapping_key: WrappingKey,
-    /// Wrapped key
-    wrapped_key: WrappedKey,
+/// Can be used to abstract wrapped key over different incoming key sizes.
+pub struct WrappedKey<T>
+where
+    T: KeySizeUser,
+    Sum<T::KeySize, U8>: ArraySize,
+    <T as KeySizeUser>::KeySize: Add<U8>,
+{
+    inner: Array<u8, Sum<T::KeySize, U8>>,
 }
-impl KeyWrapper {
-    /// Initialize a new KeyWrapper based on `KeyWrapAlgorithm` and key-to-wrap size.
-    pub(in crate::builder) fn try_new(kw_algo: &KeyWrapAlgorithm, key_size: usize) -> Result<Self> {
-        let wrapped_key = WrappedKey::try_from(key_size)?;
-        let wrapping_key = WrappingKey::from(kw_algo);
-
-        Ok(Self {
-            wrapping_key,
-            wrapped_key,
-        })
-    }
-    /// Wraps a given key.
-    ///
-    /// This function attempts to wrap the provided `target_key`.
-    ///
-    /// # Arguments
-    /// * `target_key` - A slice of bytes representing the key to be wrapped.
-    pub(in crate::builder) fn try_wrap(&mut self, target_key: &[u8]) -> Result<()> {
-        match self.wrapping_key {
-            WrappingKey::Aes128(wrap_key) => Kek::from(wrap_key)
-                .wrap(target_key, self.wrapped_key.as_mut())
-                .map_err(|_| {
-                    Error::Builder(String::from(
-                        "could not wrap key with Aes128 key wrap algorithm",
-                    ))
-                }),
-            WrappingKey::Aes192(kek) => Kek::from(kek)
-                .wrap(target_key, self.wrapped_key.as_mut())
-                .map_err(|_| {
-                    Error::Builder(String::from(
-                        "could not wrap key with Aes192 key wrap algorithm",
-                    ))
-                }),
-            WrappingKey::Aes256(kek) => Kek::from(kek)
-                .wrap(target_key, self.wrapped_key.as_mut())
-                .map_err(|_| {
-                    Error::Builder(String::from(
-                        "could not wrap key with Aes256 key wrap algorithm",
-                    ))
-                }),
-        }
+
+impl<T> WrappedKey<T>
+where
+    T: KeySizeUser,
+    Sum<T::KeySize, U8>: ArraySize,
+    <T as KeySizeUser>::KeySize: Add<U8>,
+{
+    // Function to get a mutable reference to the inner array
+    pub(crate) fn as_mut(&mut self) -> &mut [u8] {
+        self.inner.as_mut_slice()
```

Comment:
isn't that redundant with the `impl<T> AsMut<[u8]> for WrappedKey<T>` just below?

Reply:
indeed, fixed in https://github.com/nemynm/formats/pull/3/commits/ffad593a2ebeb7cebb7cc867d27aa1945366a073